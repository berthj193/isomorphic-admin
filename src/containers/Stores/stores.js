import React, { Component } from 'react';
import PropTypes from 'prop-types';
import autoBind from 'react-autobind';
import columns from './columns';
import storeQueries from '../../graphql/stores';
import withStoresManager from '../../components/apolloProviders/withStoresManager';
import ItemList from '../../components/list/item-list';

import Details from './details';
import DeleteButton from '../../components/uielements/deleteButton';

export class Stores extends Component {
  static propTypes = {
    stores: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        warehouse: PropTypes.shape({ name: PropTypes.string }),
        company: PropTypes.shape({ name: PropTypes.string }),
        channel: PropTypes.shape({ name: PropTypes.string }),
        store_products: PropTypes.arrayOf(PropTypes.object),
        created_at: PropTypes.string,
      }),
    ),
    error: PropTypes.string,
    mutate: PropTypes.func.isRequired,
  };

  static defaultProps = {
    stores: [],
  };


  constructor() {
    super();
    autoBind(this);
  }

  dataSource() {
    return this.props.stores.map(({
      id, key, created_at, store_products = [],
      warehouse = {}, company = {}, channel = {},
    }) => ({
      key,
      id,
      warehouse_name: warehouse.name,
      company_name: company.name,
      channel_name: channel.name,
      created_at,
      products_count: store_products.length,
    }));
  }

  detailsPath = ({ pathname, record: { key } }) =>
    `${pathname}/${key}`;

  deleteStore = id => e => {
    e.preventDefault();
    e.stopPropagation();
    this.props.mutate({
      mutation: storeQueries.destoryStore,
      variables: {
        id,
      },
      refetchQueries: [
        {
          query: storeQueries.getList,
        },
      ],
      messages: ['Store has been removed successfully'],
    });
  };

  actionColumn = ({ id }) => (
    <ul className="table-action">
      <li>
        <DeleteButton
          tooltipText="Remove a store"
          onConfirm={this.deleteStore(id)}
        />
      </li>
    </ul>
  );

  columns = [
    ...columns,
    {
      title: 'Action',
      dataIndex: '',
      key: 'a',
      render: this.actionColumn,
    },
  ];

  render() {
    return (
      <ItemList
        data={this.dataSource()}
        error={this.props.error}
        title="Stores"
        columns={this.columns}
        detailsComponent={Details}
        buttons={[{
          title: 'New form',
          link: '/dashboard/stores/new',
        }]}
        detailsPath={this.detailsPath}
      />
    );
  }
}

export default withStoresManager(Stores,
  {
    query: storeQueries.getList,
  }
);
