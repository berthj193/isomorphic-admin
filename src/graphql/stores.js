import gql from 'graphql-tag';

const storeReturnData = `
    id
    key
    options
    created_at
    channel {
      id
      name
    }
    warehouse {
      id
      name
    }
    legal_entity {
      id
      name
    }
`;

export const storesAutocompleteQuery = `
query {
  stores {
    id
    key
  }
}`;

export const storeListQuery = `
query {
  stores {
    ${storeReturnData}
  }
}`;

export const storeDetailsQuery = `
query stores (
  $store_key: String!
) {
  stores(
    store_key: $store_key
  ) {
    ${storeReturnData}
  }
}`;

export const createStoreQuery = `
  mutation createStore(
    $key: String!
    $options: JsonType!
    $legal_entity_id: ID!
    $warehouse_id: ID!
    $channel_id: ID!
  ) {
    createStore(
      key: $key
      options: $options
      legal_entity_id: $legal_entity_id
      warehouse_id: $warehouse_id
      channel_id: $channel_id
    )
    {
      ${storeReturnData}
    }
  }
`;

export const updateStoreQuery = `
  mutation updateStore(
    $id: ID!
    $key: String!
    $options: JsonType!
    $legal_entity_id: ID!
    $warehouse_id: ID!
    $channel_id: ID!
  ) {
    updateStore(
      id: $id
      key: $key
      options: $options
      legal_entity_id: $legal_entity_id
      warehouse_id: $warehouse_id
      channel_id: $channel_id
    )
    {
      ${storeReturnData}
    }
  }
`;

export const destroyStoreQuery = `
mutation destroyStore (
  $id: ID!
) {
  destroyStore (
    id: $id
  )
}`;

export default {
  getDetails: gql`${storeDetailsQuery}`,
  getList: gql`${storeListQuery}`,
  createStore: gql`${createStoreQuery}`,
  updateStore: gql`${updateStoreQuery}`,
  destoryStore: gql`${destroyStoreQuery}`,
  storesAutocomplete: gql`${storesAutocompleteQuery}`,
};
