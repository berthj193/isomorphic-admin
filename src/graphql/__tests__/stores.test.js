import { storeDetailsQuery, storeListQuery, createStoreQuery, updateStoreQuery } from '../stores';

describe('stores gql', () => {
  describe('storeListQuery', () => {
    it('should return proper gql request', () =>
      expect(storeListQuery).toMatchSnapshot()
    );
  });

  describe('getDetailsQuery()', () => {
    it('should return proper gql request', () =>
      expect(storeDetailsQuery).toMatchSnapshot()
    );
  });

  describe('updateStoreQuery', () => {
    it('should return proper gql request', () =>
      expect(updateStoreQuery).toMatchSnapshot()
    );
  });

  describe('createStoreQuery', () => {
    it('should return proper gql request', () =>
      expect(createStoreQuery).toMatchSnapshot()
    );
  });
});
