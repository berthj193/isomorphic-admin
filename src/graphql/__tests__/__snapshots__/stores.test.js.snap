// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stores gql createStoreQuery should return proper gql request 1`] = `
"
  mutation createStore(
    $key: String!
    $options: JsonType!
    $legal_entity_id: ID!
    $warehouse_id: ID!
    $channel_id: ID!
  ) {
    createStore(
      key: $key
      options: $options
      legal_entity_id: $legal_entity_id
      warehouse_id: $warehouse_id
      channel_id: $channel_id
    )
    {
      
    id
    key
    options
    created_at
    channel {
      id
      name
    }
    warehouse {
      id
      name
    }
    legal_entity {
      id
      name
    }

    }
  }
"
`;

exports[`stores gql getDetailsQuery() should return proper gql request 1`] = `
"
query stores (
  $store_key: String!
) {
  stores(
    store_key: $store_key
  ) {
    
    id
    key
    options
    created_at
    channel {
      id
      name
    }
    warehouse {
      id
      name
    }
    legal_entity {
      id
      name
    }

  }
}"
`;

exports[`stores gql storeListQuery should return proper gql request 1`] = `
"
query {
  stores {
    
    id
    key
    options
    created_at
    channel {
      id
      name
    }
    warehouse {
      id
      name
    }
    legal_entity {
      id
      name
    }

  }
}"
`;

exports[`stores gql updateStoreQuery should return proper gql request 1`] = `
"
  mutation updateStore(
    $id: ID!
    $key: String!
    $options: JsonType!
    $legal_entity_id: ID!
    $warehouse_id: ID!
    $channel_id: ID!
  ) {
    updateStore(
      id: $id
      key: $key
      options: $options
      legal_entity_id: $legal_entity_id
      warehouse_id: $warehouse_id
      channel_id: $channel_id
    )
    {
      
    id
    key
    options
    created_at
    channel {
      id
      name
    }
    warehouse {
      id
      name
    }
    legal_entity {
      id
      name
    }

    }
  }
"
`;
